bin/demo.py

```diff
@@ -268,30 +275,34 @@ def uninstall(dcos_url, jenkins_name):
     jenkins_url = '{}service/{}/'.format(dcos_url, jenkins_name)
 
     config_dcos_cli(dcos_url)
-    check_and_set_token(jenkins_url)
+    check_and_set_token()
 
     try:
         if arguments['install']:
-            make_temp_dir()
-            if not verify(jenkins_url):
-                install(dcos_url, jenkins_name, jenkins_url)
-            remove_temp_dir()
-        elif arguments['pipeline']:
+            if not verify_jenkins(jenkins_url):
+                log("couldn't find Jenkins running at '{}'".format(jenkins_url))
+                install_jenkins(jenkins_name, jenkins_url)
+        elif arguments['pipeline'] or arguments['pipeline-old']:
             branch = arguments['--branch'].lower()
             if branch == 'master':
-                log_and_exit ("Cannot run demo against the master branch.")
+                log_and_exit("!! cannot run demo against the master branch.")
             org = arguments['--org']
             username = arguments['--username']
             password = arguments['--password']
-            check_marathon_lb(elb_url)
-            update_and_push_marathon_json(elb_url, branch)
-            demo_pipeline(jenkins_url, dcos_url, elb_url, jenkins_name, branch, org, username, password)
+#            if not verify_marathon_lb(elb_url):
+#                log("couldn't find marathon-lb running at '{}'".format(elb_url))
+#                install_marathon_lb(elb_url)
+#            update_and_push_marathon_json(elb_url, branch)
+            if arguments['pipeline']:
```

Comment:
Maybe check for `pipeline-old` to allow toggling and default to `pipeline` workflow if `pipeline-old` is not specified.